### PR TITLE
fix: the API request payload to not omitempty properties when sending event_rule update requests to upstream

### PR DESCRIPTION
**Pull request by abhimanyubabbar**
## Description
We were omitting empty properties when attaching events to trackingplan. The backend expects the properties array to be present on the payload even though empty when attaching event_rule on trackingplan.

The changes in fix are as follows:

1. This fix prevents omitting empty properties.
2. Add corresponding e2e test for attaching event with empty properties on the trackingplan.

### CODE REVIEW

**Review by dhawal1248 — APPROVED** [review]

File: cli/tests/testdata/expected/upstream/create/tracking-plan_api_tracking [hunk 1]
```diff
@@ -115,11 +115,25 @@
...
+    },
+    {
+      "categoryId": null,
+      "createdAt": "2025-07-21T09:09:51.019Z",
+      "createdBy": "300EdpkzmrxHDc0k8nuVsZdS2t3",
+      "description": "This event is triggered when a user logs out.",
+      "eventType": "track",
+      "id": "ev_30B7DOLrwGBY2MjnwJPKskp43eT",
+      "identitySection": "properties",
+      "name": "Logout Event",
```
[the remark this answers is not in the document]
Comment (abhimanyubabbar) [reply cli/tests/testdata/expected/upstream/create/tracking-plan_api_tracking:127]: The request payload for the trackingplan event identifiers also uses omitempty flag when reading the upstream trackingplan event properties
